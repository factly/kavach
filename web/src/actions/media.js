import axios from 'axios';
import {
  ADD_MEDIUM,
  ADD_MEDIA,
  ADD_MEDIA_REQUEST,
  SET_MEDIA_LOADING,
  RESET_MEDIA,
  MEDIA_API,
} from '../constants/media';
import { buildObjectOfItems } from '../utils/objects';

import { addErrorNotification, addSuccessNotification } from './notifications';

export const getMedia = (query) => {
  return (dispatch) => {
    dispatch(loadingMedia());
    return axios
      .get(MEDIA_API, {
        params: query,
      })
      .then((response) => {
        dispatch(addMediaList(response.data.nodes));
        dispatch(
          addMediaRequest({
            data: response.data.nodes.map((item) => item.id),
            query: query,
            total: response.data.total,
          }),
        );
        dispatch(stopMediaLoading());
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const getMedium = (id) => {
  return (dispatch) => {
    dispatch(loadingMedia());
    return axios
      .get(MEDIA_API + '/' + id)
      .then((response) => {
        dispatch(getMediumByID(response.data));
        dispatch(stopMediaLoading());
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const addMedium = (data) => {
  return (dispatch) => {
    dispatch(loadingMedia());
    return axios
      .post(MEDIA_API, data)
      .then(() => {
        dispatch(resetMedia());
        dispatch(addSuccessNotification('Media Added'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const updateMedium = (data) => {
  return (dispatch) => {
    dispatch(loadingMedia());
    return axios
      .put(MEDIA_API + '/' + data.id, data)
      .then((response) => {
        dispatch(getMediumByID(response.data));
        dispatch(stopMediaLoading());
        dispatch(addSuccessNotification('Media Updated'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const deleteMedium = (id) => {
  return (dispatch) => {
    dispatch(loadingMedia());
    return axios
      .delete(MEDIA_API + '/' + id)
      .then(() => {
        dispatch(resetMedia());
        dispatch(addSuccessNotification('Media Deleted'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const addMedia = (media) => {
  return (dispatch) => {
    dispatch(addMediaList(media));
  };
};

export const loadingMedia = () => ({
  type: SET_MEDIA_LOADING,
  payload: true,
});

export const stopMediaLoading = () => ({
  type: SET_MEDIA_LOADING,
  payload: false,
});

export const getMediumByID = (data) => ({
  type: ADD_MEDIUM,
  payload: data,
});

export const addMediaList = (data) => (dispatch) => {
  dispatch({
    type: ADD_MEDIA,
    payload: buildObjectOfItems(data),
  });
};

export const addMediaRequest = (data) => ({
  type: ADD_MEDIA_REQUEST,
  payload: data,
});

export const resetMedia = () => ({
  type: RESET_MEDIA,
});
