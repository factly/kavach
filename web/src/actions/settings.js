import { TOGGLE_SIDER } from '../constants/settings';

export const toggleSider = () => {
  return (dispatch, getState) => {
    dispatch({
      type: TOGGLE_SIDER,
    });
  };
};
