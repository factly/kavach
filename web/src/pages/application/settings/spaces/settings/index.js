import React from 'react';
import { useParams } from 'react-router-dom';
import SettingsList from '../../../../../components/Settings';

export default function SpaceSettings() {
  const { appID, spaceID } = useParams();
  return <SettingsList type="space" appID={appID} spaceID={spaceID} />;
}
