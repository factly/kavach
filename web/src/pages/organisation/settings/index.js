import React from 'react';
import { useSelector } from 'react-redux';
import SettingsList from '../../../components/Settings';
import { Skeleton } from 'antd';
export default function OrganisationSettings({ orgID }) {
  const { role, loading } = useSelector((state) => {
    return {
      role: state.profile.roles[state.organisations.selected],
      loading: state.profile.loading,
    };
  });
  return (
    <div>
      {
      (loading) ? <Skeleton/> :<SettingsList type="organisation" orgID={orgID} role={role}/>
      }
    </div>
  );
}
