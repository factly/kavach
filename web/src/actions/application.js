import axios from 'axios';
import {
  ADD_APPLICATION,
  ADD_APPLICATIONS,
  ADD_APPLICATIONS_REQUEST,
  SET_APPLICATIONS_LOADING,
  RESET_APPLICATIONS,
  APPLICATIONS_API,
} from '../constants/application';
import { addMediaList } from './media';
import { addErrorNotification, addSuccessNotification } from './notifications';

export const getApplications = () => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
      return axios
        .get(APPLICATIONS_API + '/' + getState().organisations.selected + '/applications')
        .then((response) => {
          dispatch(
            addMediaList(
              response.data
                .filter((application) => application.medium)
                .map((application) => application.medium),
            ),
          );
          dispatch(
            addApplicationsList(
              response.data.map((application) => {
                return { ...application, medium: application.medium?.id };
              }),
            ),
          );
          dispatch(
            addApplicationsRequest({
              data: response.data.map((item) => item.id),
             
              total: response.data.total,
            }),
          );
          dispatch(stopApplicationLoading());  
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const addDefaultApplications = () => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
      return axios
        .post(APPLICATIONS_API + '/' + getState().organisations.selected + '/applications/default')
        .then((response) => {
          dispatch(
            addMediaList(
              response.data
                .filter((application) => application.medium)
                .map((application) => application.medium),
            ),
          );
          dispatch(
            addApplicationsList(
              response.data.map((application) => {
                return { ...application, medium: application.medium?.id };
              }),
            ),
          );
          dispatch(
            addApplicationsRequest({
              data: response.data.map((item) => item.id),
              total: response.data.total,
            }),
          );
          dispatch(stopApplicationLoading()); 
          dispatch(addSuccessNotification('Factly Applications Added')) 
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const getApplication = (id) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .get(APPLICATIONS_API + '/' + getState().organisations.selected + '/applications/' + id)
      .then((response) => {
        if( response.data.medium) dispatch(addMediaList([response.data.medium]));
          dispatch(getApplicationByID({ ...response.data, medium: response.data.medium?.id }));
          dispatch(stopApplicationLoading());
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const addApplication = (data) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .post(APPLICATIONS_API + '/' + getState().organisations.selected + '/applications', data)
      .then(() => {
        dispatch(resetApplications());
        dispatch(addSuccessNotification('Application Added')) 
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const updateApplication = (data) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
    .put(APPLICATIONS_API + '/' + getState().organisations.selected + '/applications/' + data.id, data)
    .then((response) => {
      if (response.data.medium) dispatch(addMediaList([response.data.medium]));
        dispatch(getApplicationByID({ ...response.data, medium: response.data.medium?.id }));
        dispatch(stopApplicationLoading());
        dispatch(addSuccessNotification('Application Updated')) 
    })
    .catch((error) => {
      dispatch(addErrorNotification(error.message));
    });
  };
};

export const deleteApplication = (id) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .delete(APPLICATIONS_API + '/' + getState().organisations.selected + '/applications/' + id)
      .then(() => {
        dispatch(resetApplications());
        dispatch(addSuccessNotification('Application Deleted')) 
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const addApplications = (applications) => {
  return (dispatch) => {
    dispatch(
      addMediaList(
        applications.filter((application) => application.medium).map((application) => application.medium),
      ),
    );
    dispatch(
      addApplicationsList(
        applications.map((application) => {
          return { ...application, medium: application.medium?.id};
        }),
      ),
    );
  };
};

export const loadingApplications = () => ({
  type: SET_APPLICATIONS_LOADING,
  payload: true,
});

export const stopApplicationLoading = () => ({
  type: SET_APPLICATIONS_LOADING,
  payload: false,
});

export const getApplicationByID = (data) => ({
  type: ADD_APPLICATION,
  payload: data,
});

export const addApplicationsList = (data) => ({
  type: ADD_APPLICATIONS,
  payload: data,
});

export const addApplicationsRequest = (data) => ({
  type: ADD_APPLICATIONS_REQUEST,
  payload: data,
});

export const resetApplications = () => ({
  type: RESET_APPLICATIONS,
});