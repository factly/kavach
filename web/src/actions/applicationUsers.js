import axios from 'axios';
import {
  ADD_APPLICATION_USER,
  ADD_APPLICATION_USERS,
  ADD_APPLICATION_USERS_REQUEST,
  SET_APPLICATION_USERS_LOADING,
  RESET_APPLICATION_USERS,
  APPLICATION_USERS_API,
  ADD_USER_IDS,
} from '../constants/applicationUser';
import { getIds } from '../utils/objects';
import { addMediaList } from './media';
import { addErrorNotification, addSuccessNotification } from './notifications';
import { loadingUsers, stopUsersLoading } from './users';

export const getApplications = (appID) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .get(
        APPLICATION_USERS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          appID +
          '/users',
      )
      .then((response) => {
        dispatch(
          addApplicationList(
            response.data.users.map((user) => {
              return user;
            }),
          ),
        );
        dispatch(
          addApplicationsRequest({
            data: response.data.users.map((item) => item.id),
            application_id: response.data.application.id,
            total: response.data.total,
          }),
        );
        dispatch(stopApplicationLoading());
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const getApplication = (id) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .get(
        APPLICATION_USERS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          id +
          '/users',
      )
      .then((response) => {
        if (response.data.medium) dispatch(addMediaList([response.data.medium]));
        dispatch(getApplicationByID({ ...response.data, medium: response.data.medium?.id }));
        dispatch(stopApplicationLoading());
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const addApplicationUser = (data) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .post(
        APPLICATION_USERS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          data.application_id +
          '/users',
        data,
      )
      .then(() => {
        dispatch(resetApplications());
        dispatch(addSuccessNotification('Application User Added'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const updateApplication = (data) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .put(
        APPLICATION_USERS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          data.id,
        data,
      )
      .then((response) => {
        if (response.data.medium) dispatch(addMediaList([response.data.medium]));
        dispatch(getApplicationByID({ ...response.data, medium: response.data.medium?.id }));
        dispatch(stopApplicationLoading());
        dispatch(addSuccessNotification('Application Updated'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const deleteApplication = (id, appID) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .delete(
        APPLICATION_USERS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          id +
          '/users/' +
          appID,
      )
      .then(() => {
        dispatch(resetApplications());
        dispatch(addSuccessNotification('Application User Deleted'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const getApplicationUsers = (appID) => {
  return (dispatch, getState) => {
    dispatch(loadingUsers());
    return axios
      .get(
        APPLICATION_USERS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          appID +
          '/users',
      )
      .then((res) => {
        dispatch(addUserIds(getIds(res.data.users), appID));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      })
      .finally(() => {
        dispatch(stopUsersLoading());
      });
  };
};

export const addApplications = (applications) => {
  return (dispatch) => {
    dispatch(
      addMediaList(
        applications
          .filter((application) => application.medium)
          .map((application) => application.medium),
      ),
    );
    dispatch(
      addApplicationList(
        applications.map((application) => {
          return { ...application, medium: application.medium?.id };
        }),
      ),
    );
  };
};

export const loadingApplications = () => ({
  type: SET_APPLICATION_USERS_LOADING,
  payload: true,
});

export const stopApplicationLoading = () => ({
  type: SET_APPLICATION_USERS_LOADING,
  payload: false,
});

export const getApplicationByID = (data) => ({
  type: ADD_APPLICATION_USER,
  payload: data,
});

export const addApplicationList = (data) => ({
  type: ADD_APPLICATION_USERS,
  payload: data,
});

export const addApplicationsRequest = (data) => ({
  type: ADD_APPLICATION_USERS_REQUEST,
  payload: data,
});

const addUserIds = (data, appID) => ({
  type: ADD_USER_IDS,
  payload: {
    id: appID,
    data: data,
  },
});
export const resetApplications = () => ({
  type: RESET_APPLICATION_USERS,
});
