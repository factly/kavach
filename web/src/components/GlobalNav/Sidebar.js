import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { Layout, Menu, Button } from 'antd';
import routes from '../../config/routes';
import { setCollapse } from './../../actions/sidebar';
import { MenuUnfoldOutlined, MenuFoldOutlined } from '@ant-design/icons';
const { Sider } = Layout;

function Sidebar() {
  const {
    // sider: { collapsed },
    navTheme,
  } = useSelector((state) => state.settings);
  const dispatch = useDispatch();
  const { collapsed } = useSelector((state) => state.sidebar);
  const onCollapse = (collapsed) => {
    collapsed ? dispatch(setCollapse(true)) : dispatch(setCollapse(false));
  };
  const buttonStyle = {
    borderRadius: '50px',
    padding: '0.25rem 0.5rem',
  };
  return (
    <Sider
      breakpoint="xl"
      className="main-sidebar"
      width="256"
      theme={navTheme}
      collapsible
      collapsed={collapsed}
      onCollapse={onCollapse}
      collapsedWidth={48}
      trigger={null}
      style={{
        position: 'sticky',
        background: '#f0f2f5',
        left: 0,
        top: 0,
        overflow: 'auto',
        height: '100vh',
      }}
    >
      <Link to="/settings">
        <div className="menu-header">
          <img
            alt="logo"
            hidden={!collapsed}
            className="menu-logo"
            src={require('../../assets/kavach_icon.png')}
          />
          <img
            alt="logo"
            hidden={collapsed}
            src={require('../../assets/kavach.png')}
            style={{ width: '60%' }}
          />
        </div>
      </Link>
      <Menu
        theme={navTheme}
        mode="inline"
        className="slider-menu"
        style={{ background: '#f0f2f5' }}
      >
        {routes
          .filter((each) => each.enableNavigation === true)
          .map((route, index) => {
            const { Icon } = route;
            return (
              <Menu.Item key={index}>
                <Link to={route.path}>
                  <Icon></Icon>
                  <span>{route.title}</span>
                </Link>
              </Menu.Item>
            );
          })}
      </Menu>
      {!collapsed ? (
        <div
          style={{
            display: 'flex',
            justifyContent: 'space-between',
            padding: '12px 24px',
            lineHeight: '40px',
            alignItems: 'center',
            width: '100%',
            position: 'absolute',
            bottom: '0',
            background: '#f0f2f5',
          }}
        >
          <div>
            <Button
              style={{ ...buttonStyle, marginLeft: '0.25rem' }}
              onClick={() => onCollapse(true)}
            >
              <MenuFoldOutlined />
            </Button>
          </div>
        </div>
      ) : (
        <div
          style={{
            display: 'flex',
            justifyContent: 'space-between',
            padding: '12px 8px',
            lineHeight: '40px',
            alignItems: 'center',
            width: '100%',
            position: 'absolute',
            bottom: '0',
            background: '#f0f2f5',
          }}
        >
          <Button style={buttonStyle} onClick={() => onCollapse(false)}>
            <MenuUnfoldOutlined />
          </Button>
        </div>
      )}
    </Sider>
  );
}

export default Sidebar;
