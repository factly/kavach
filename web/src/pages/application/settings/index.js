import React from 'react';
import { Link, useParams } from 'react-router-dom';
import SettingsList, { SettingsCard } from '../../../components/Settings';
import { Col, Row, Skeleton, Descriptions, Divider, Space } from 'antd';
import { ApartmentOutlined } from '@ant-design/icons';
import { getApplication } from '../../../actions/application';
import { useDispatch, useSelector } from 'react-redux';

export default function ApplicationSettings() {
  const { id } = useParams();
  const dispatch = useDispatch();
  const descriptionSpan = 3;
  const { application, loadingApp, role, loading } = useSelector((state) => {
    return {
      application: state.applications.details[id],
      loadingApp: state.applications.loading,
      role: state.profile.roles[state.organisations.selected],
      loading: state.profile.loading,
    };
  });

  React.useEffect(() => {
    dispatch(getApplication(id));
    //eslint-disable-next-line
  }, [id]);

  return (
    <div>
      {loading || loadingApp ? (
        <Skeleton />
      ) : (
        <Space direction="vertical" style={{ width: '100%' }}>
          <Descriptions title={<h2> Manage Application </h2>} bordered={true}>
            <Descriptions.Item label="Name" span={descriptionSpan}>
              {application?.name}
            </Descriptions.Item>
            <Descriptions.Item label="Description" span={descriptionSpan}>
              {application?.description}
            </Descriptions.Item>
            <Descriptions.Item label="Number of Spaces" span={descriptionSpan}>
              {application?.spaces?.length}
            </Descriptions.Item>
            <Descriptions.Item label="Number of Tokens" span={descriptionSpan}>
              {application?.tokens?.length}
            </Descriptions.Item>
            <Descriptions.Item label="Number of Tokens" span={descriptionSpan}>
              <a href={`https://${application?.url}`}>{application?.url}</a>
            </Descriptions.Item>
          </Descriptions>
          <Divider> Application Settings</Divider>
          <SettingsList type="application" appID={id} role={role} />
          <Row style={{ marginTop: 16 }}>
            <Col span={12}>
              <Link to={`/applications/${id}/settings/spaces`}>
                <SettingsCard
                  icon={<ApartmentOutlined style={{ color: '#4E89FF' }} />}
                  title="Spaces"
                  description="Space Settings"
                />
              </Link>
            </Col>
          </Row>
        </Space>
      )}
    </div>
  );
}
