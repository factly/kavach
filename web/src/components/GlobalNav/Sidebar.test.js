import React from 'react';
import { useHistory } from 'react-router-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { useDispatch, Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount, shallow } from 'enzyme';

import '../../matchMedia.mock';
import Sidebar from './Sidebar';
import * as actions from '../../actions/settings';
import { Layout } from 'antd';

const { Sider } = Layout;

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
}));

jest.mock('../../actions/settings', () => ({
  toggleSider: jest.fn(),
}));

describe('Sidebar component', () => {
  let store;
  let mockedDispatch;
  let state = {
    settings: {
      navTheme: 'dark',
      title: 'Dega',
      sider: {
        collapsed: true,
      },
    },
  };
  store = mockStore(() => state);
  store.dispatch = jest.fn(() => ({}));
  mockedDispatch = jest.fn(() => Promise.resolve({}));
  useDispatch.mockReturnValue(mockedDispatch);
  describe('snapshot testing', () => {
    it('should render the component', () => {
      const tree = shallow(
        <Provider store={store}>
          <Router>
            <Sidebar />
          </Router>
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper;
    it('should call toggleSider', (done) => {
      actions.toggleSider.mockReset();
      const push = jest.fn();
      useHistory.mockReturnValueOnce({ push });
      store = mockStore(state);

      wrapper = mount(
        <Provider store={store}>
          <Router>
            <Sidebar />
          </Router>
        </Provider>,
      );

      wrapper.find(Sider).props().onBreakpoint();
      wrapper.update();

      expect(actions.toggleSider).toHaveBeenCalledWith();
      done();
    });
  });
});
