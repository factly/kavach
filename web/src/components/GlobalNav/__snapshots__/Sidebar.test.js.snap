// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component snapshot testing should render the component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [MockFunction],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [MockFunction],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <BrowserRouter>
    <Sidebar />
  </BrowserRouter>
</ContextProvider>
`;
